server/src/test/java/io/crate/execution/dml/IndexerTest.java

```diff
@@ -816,6 +816,21 @@ public void test_array_length_is_indexed() throws Exception {
         assertTranslogParses(doc, table);
     }
 
+    @Test
+    public void test_only_top_most_array_length_is_indexed_for_multi_dimensional_arrays() throws Exception {
+        SQLExecutor e = SQLExecutor.of(clusterService)
+            .addTable("create table tbl (xs int[][])");
+        DocTableInfo table = e.resolveTableInfo("tbl");
+
+        var indexer = getIndexer(e, "tbl", "xs");
+        ParsedDocument doc = indexer.index(item(List.of(List.of(1), List.of(1, 1), List.of(1, 2, 3, 4))));
+        IndexableField[] arrayLengthFields = doc.doc().getFields(toArrayLengthFieldName((Reference) e.asSymbol("xs"), table::getReference));
+
+        assertThat(arrayLengthFields.length).isEqualTo(1);
+        assertThat(arrayLengthFields[0].toString()).isEqualTo("IntField <_array_length_1:3>");
```

Comment:
Otherwise we would have
```
-- top level
IntField <_array_length_1:3>
-- inner
IntField <_array_length_1:1>
IntField <_array_length_1:2>
IntField <_array_length_1:4>
```
indexed.